packages/keyring-eth-ledger-bridge/src/ledger-error-handler.test.ts

```diff
@@ -0,0 +1,170 @@
+import { TransportStatusError } from '@ledgerhq/hw-transport';
+
+import { handleLedgerTransportError } from './ledger-error-handler';
+import { LedgerStatusError } from './type';
+
+describe('handleLedgerTransportError', () => {
+  const fallbackMessage = 'Default error message';
+
+  /**
+   * Helper function to create a TransportStatusError-like object
+   *
+   * @param message - The error message
+   * @param statusCode - The status code
+   * @returns A TransportStatusError instance
+   */
+  function createTransportStatusError(
+    message: string,
+    statusCode: number,
+  ): TransportStatusError {
+    const error = {
+      statusCode,
+      message,
+      name: 'TransportStatusError',
+    };
+    Object.setPrototypeOf(error, TransportStatusError.prototype);
+    return error as TransportStatusError;
+  }
+
+  /**
+   * Helper function to test that handleLedgerTransportError throws a LedgerStatusError
+   * with expected properties
+   *
+   * @param error - The error to pass to handleLedgerTransportError
+   * @param expectedStatusCode - Expected status code of the thrown LedgerStatusError
+   * @param expectedMessage - Expected message of the thrown LedgerStatusError
+   */
+  function expectLedgerStatusError(
+    error: unknown,
+    expectedStatusCode: number,
+    expectedMessage: string,
+  ): void {
+    expect(() => handleLedgerTransportError(error, fallbackMessage)).toThrow(
+      LedgerStatusError,
+    );
+
+    let thrownError: unknown;
+    try {
+      handleLedgerTransportError(error, fallbackMessage);
+    } catch (error_: unknown) {
+      thrownError = error_;
+    }
+    expect(thrownError).toBeInstanceOf(LedgerStatusError);
+    expect((thrownError as LedgerStatusError).statusCode).toBe(
+      expectedStatusCode,
+    );
+    expect((thrownError as LedgerStatusError).message).toBe(expectedMessage);
+  }
+
+  describe('when error is TransportStatusError', () => {
+    it('handles status code 0x6985 (user rejection)', () => {
+      const error = createTransportStatusError('User rejected', 0x6985);
+      expectLedgerStatusError(
+        error,
+        0x6985,
+        'Ledger: User rejected the transaction',
+      );
```

Comment:
Since there's a lint warning for each tests (because there's no `expect` call in the `it` callback. WDYT about making `expectLedgerStatusError` return `true` and use it like:


```suggestion
      expect(expectLedgerStatusError(
        error,
        0x6985,
        'Ledger: User rejected the transaction',
      )).toBe(true);
```

This is super useful, but it would at least avoid the warnings everywhere.

Other options would be to use `it.each` and combine all the tests together, so we would not need the `expectLedgerStatusError`.

Something like:
```ts
it.each([
  { tc: 'user rejection', status: 0x6985, message: 'Ledger: User rejected the transaction' },
  ...
]('handles status code $status ($tc)', ({ status, error } => {
  ...
});
```